include/deal.II/lac/linear_operator.h

```diff
@@ -821,6 +914,114 @@ namespace internal
         v.reinit(matrix.n(), omit_zeroing_entries);
       }
     };
+
+
+    /**
+    * A dummy class for LinearOperators that do not require any extensions
+    * to facilitate the operations of the matrix.
+    *
+    * This is the Payload class typically associated with deal.II's native
+    * SparseMatrix. To use Trilinos and PETSc SparseMatrices it is necessary
+    * to initialize a LinearOperator with their associated Payload.
+    *
+    * @author Jean-Paul Pelteret, 2016
+    *
+    * @ingroup LAOperators
+    */
+    class EmptyPayload
+    {
+    public:
+
+      /**
+      * Default constructor
+      */
+      EmptyPayload ()
```

Comment:
As a sidenote: you define a number of constructors here, but they all don't do very much. You probably don't care about matching any particular interface, so why not just do something like
```
  template <typename... Args>
  EmptyPayload (Args...) {}  // eat zero or more arguments, of any type
```